weave/trace_server/calls_query_builder.py

```diff
@@ -998,6 +1012,139 @@ def make_call_parts_predicate_filters_sql(
+    pb: ParamBuilder,
+    table_alias: str,
+) -> Optional[str]:
+    """Creates a LIKE-optimized condition for equality operations."""
+    # Check if the left side is a GetField operation on a JSON field
+    if not isinstance(operation.eq_[0], tsi_query.GetFieldOperator):
+        return None
+
+    # Check if the right side is a literal string
+    if not isinstance(operation.eq_[1], tsi_query.LiteralOperation) or not isinstance(
+        operation.eq_[1].literal_, str
+    ):
+        return None
+
+    field = get_field_by_name(operation.eq_[0].get_field_).field
+    literal_value = operation.eq_[1].literal_
+    like_pattern = f'%"{literal_value}"%'
+
+    return _create_like_condition(field, like_pattern, pb, table_alias)
+
+
+def _create_like_optimized_contains_condition(
+    operation: tsi_query.ContainsOperation,
+    condition: Condition,
+    pb: ParamBuilder,
+    table_alias: str,
+) -> Optional[str]:
+    """Creates a LIKE-optimized condition for contains operations."""
+    # Check if the input is a GetField operation on a JSON field
+    if not isinstance(operation.contains_.input, tsi_query.GetFieldOperator):
+        return None
+
+    # Check if the substring is a literal string
+    if not isinstance(
+        operation.contains_.substr, tsi_query.LiteralOperation
+    ) or not isinstance(operation.contains_.substr.literal_, str):
+        return None
+
+    field = get_field_by_name(operation.contains_.input.get_field_).field
+    substr_value = operation.contains_.substr.literal_
+    case_insensitive = operation.contains_.case_insensitive or False
+    like_pattern = f'%"%{substr_value}%"%'
+
+    return _create_like_condition(
+        field, like_pattern, pb, table_alias, case_insensitive
+    )
+
+
+def _create_like_optimized_in_condition(
+    operation: tsi_query.InOperation,
+    condition: Condition,
+    pb: ParamBuilder,
+    table_alias: str,
+) -> Optional[str]:
+    """Creates a LIKE-optimized condition for in operations."""
+    # Check if the left side is a GetField operation on a JSON field
+    if not isinstance(operation.in_[0], tsi_query.GetFieldOperator):
+        return None
+
+    # Check if the right side is a list of values
+    if len(operation.in_) != 2 or not isinstance(operation.in_[1], list):
+        return None
+    if len(operation.in_[1]) == 0:
+        return None
+
+    field = get_field_by_name(operation.in_[0].get_field_).field
+
+    # Create OR conditions for each value
+    like_conditions: list[str] = []
+
+    for value_operand in operation.in_[1]:
+        if not isinstance(value_operand, tsi_query.LiteralOperation) or not isinstance(
+            value_operand.literal_, str
+        ):
+            return None
+
+        value = value_operand.literal_
+        like_pattern = f'%"{value}"%'
+        param_name = pb.add_param(like_pattern)
+        field_name = f"{table_alias}.{field}"
+        like_conditions.append(f"{field_name} LIKE {_param_slot(param_name, 'String')}")
```

Comment:
why not use your _create_like_condition here?

Reply:
nice, changing. I refactored _create_like_condition and did not re-add it.